### ISSUE TITLE
Children's diseases in prototype-framework:v1.12.0

### ISSUE THREAD

**Issue by stefjoosten** [AmpersandTarski/prototype#84]
# What happened

1. I opened a new project from the repository Ampersandtarski/project-template@bb298a15171a8615014867801d17a56e76745f3f
2. I copied Enroll.adl (from Ampersandtarski/RAP/Demos/Enroll) to the new project and renamed it to MyScript.adl
3. I set passwords in .env and built the prototype with `docker compose build`
4. I cleaned out my entire docker environment locally: no volumes, no containers, no networks, nothing but images.
5. I caught the log file of the container prototype in the file `prototype.log`.
6. I spun up my prototype with `docker compose up -d` and look at `localhost` in my browser.
7. 05/Dec/2021:07:26:07	start	I saw a white body in an Ampersand frame in my browser.
8. 05/Dec/2021:07:28:54	<Enter>	I saw the installer with the red button “reinstall application”.
9. 05/Dec/2021:07:34:28	<“reinstall application”>	I saw the same screen, but now with a (red) error message: “Typed property Ampersand\Model::$compilerVersion must not be accessed before initialization”. This is visible in the log too.
10. 05/Dec/2021:07:38:19	<RELOAD>	I saw the same screen with a different message: “Cannot determine latest installed model version in MySQL database prototype-db - MyScript. Try reinstalling the database”. This message does NOT show up in the log file.
11. 05/Dec/2021:07:43:46	<“reinstall application”>	Now the installation succeeded.

The event timestamps mentioned above correspond to the 
[prototype.log](https://github.com/AmpersandTarski/Ampersand/files/7655478/prototype.log)

**Comment by Michiel-s** [comment]
The first time you didn't see anything on the screen (bullet point 7) this had to do with incorrect database credentials or the database not running. See your prototype.log at line 71. You used the password `blabla`

A few minutes later you've fixed that somehow. Maybe removing the database volume and restarting or manually setting the password for the `ampersand` user.

You tried again and got redirected to the installer page because of a missing database implementation. In debug mode this is not logged as an error, because it is the expected behavior for the first time access to the application.

The error you then mention `I saw the same screen, but now with a (red) error message: “Typed property Ampersand\Model::$compilerVersion must not be accessed before initialization”. This is visible in the log too.` SHOULD NOT be there. I cannot reproduce at this moment. When I take the prototype framework version v1.12.0 this is not happening. But in some rare case with database initiate partly maybe there is an edge case I'm overlooking.

The rest is expected behavior to notify you about possibly incompatible database structure with current Ampersand model. This is a feature we introduced for AmpersandTarski/Ampersand#1208

**Comment by hanjoosten** [comment]
I encounterd the same problem today:
![image](https://user-images.githubusercontent.com/1159672/158059700-0b86feb2-3128-4376-bd05-112902a8dfd8.png)

log:

~~~
Error/Exception
Type: Error
Message: Typed property Ampersand\Model::$compilerVersion must not be accessed before initialization
File: /var/www/src/Ampersand/Plugs/MysqlDB/MysqlDB.php
Line: 219
Trace:
#0 /var/www/src/Ampersand/AmpersandApp.php(646): Ampersand\Plugs\MysqlDB\MysqlDB->addToModelVersionHistory(Object(Ampersand\Model))
#1 /var/www/src/Ampersand/AmpersandApp.php(588): Ampersand\AmpersandApp->registerCurrentModelVersion()
#2 /var/www/public/api/v1/admin.installer.php(37): Ampersand\AmpersandApp->reinstall(true, false)
#3 [internal function]: Closure->{closure}(Object(Slim\Http\Request), Object(Slim\Http\Response), Array)
#4 /var/www/lib/slim/slim/Slim/Handlers/Strategies/RequestResponse.php(40): call_user_func(Object(Closure), Object(Slim\Http\Request), Object(Slim\Http\Response), Array)
#5 /var/www/lib/slim/slim/Slim/Route.php(281): Slim\Handlers\Strategies\RequestResponse->__invoke(Object(Closure), Object(Slim\Http\Request), Object(Slim\Http\Response), Array)
#6 /var/www/public/api/v1/index.php(198): Slim\Route->__invoke(Object(Slim\Http\Request), Object(Slim\Http\Response))
#7 [internal function]: Closure->{closure}(Object(Slim\Http\Request), Object(Slim\Http\Response), Object(Slim\Route))
#8 /var/www/lib/slim/slim/Slim/DeferredCallable.php(57): call_user_func_array(Object(Closure), Array)
#9 [internal function]: Slim\DeferredCallable->__invoke(Object(Slim\Http\Request), Object(Slim\Http\Response), Object(Slim\Route))
#10 /var/www/lib/slim/slim/Slim/MiddlewareAwareTrait.php(70): call_user_func(Object(Slim\DeferredCallable), Object(Slim\Http\Request), Object(Slim\Http\Response), Object(Slim\Route))
#11 /var/www/public/api/v1/admin.installer.php(114): Slim\Route->Slim\{closure}(Object(Slim\Http\Request), Object(Slim\Http\Response))
#12 [internal function]: Closure->{closure}(Object(Slim\Http\Request), Object(Slim\Http\Response), Object(Closure))
#13 /var/www/lib/slim/slim/Slim/DeferredCallable.php(57): call_user_func_array(Object(Closure), Array)
#14 [internal function]: Slim\DeferredCallable->__invoke(Object(Slim\Http\Request), Object(Slim\Http\Response), Object(Closure))
#15 /var/www/lib/slim/slim/Slim/MiddlewareAwareTrait.php(70): call_user_func(Object(Slim\DeferredCallable), Object(Slim\Http\Request), Object(Slim\Http\Response), Object(Closure))
#16 /var/www/lib/slim/slim/Slim/MiddlewareAwareTrait.php(117): Slim\Route->Slim\{closure}(Object(Slim\Http\Request), Object(Slim\Http\Response))
#17 /var/www/lib/slim/slim/Slim/Route.php(268): Slim\Route->callMiddlewareStack(Object(Slim\Http\Request), Object(Slim\Http\Response))
#18 /var/www/lib/slim/slim/Slim/App.php(503): Slim\Route->run(Object(Slim\Http\Request), Object(Slim\Http\Response))
#19 /var/www/public/api/v1/index.php(166): Slim\App->__invoke(Object(Slim\Http\Request), Object(Slim\Http\Response))
#20 [internal function]: Closure->{closure}(Object(Slim\Http\Request), Object(Slim\Http\Response), Object(Slim\App))
#21 /var/www/lib/slim/slim/Slim/DeferredCallable.php(57): call_user_func_array(Object(Closure), Array)
#22 [internal function]: Slim\DeferredCallable->__invoke(Object(Slim\Http\Request), Object(Slim\Http\Response), Object(Slim\App))
#23 /var/www/lib/slim/slim/Slim/MiddlewareAwareTrait.php(70): call_user_func(Object(Slim\DeferredCallable), Object(Slim\Http\Request), Object(Slim\Http\Response), Object(Slim\App))
#24 /var/www/public/api/v1/index.php(220): Slim\App->Slim\{closure}(Object(Slim\Http\Request), Object(Slim\Http\Response))
#25 [internal function]: Closure->{closure}(Object(Slim\Http\Request), Object(Slim\Http\Response), Object(Closure))
#26 /var/www/lib/slim/slim/Slim/DeferredCallable.php(57): call_user_func_array(Object(Closure), Array)
#27 [internal function]: Slim\DeferredCallable->__invoke(Object(Slim\Http\Request), Object(Slim\Http\Response), Object(Closure))
#28 /var/www/lib/slim/slim/Slim/MiddlewareAwareTrait.php(70): call_user_func(Object(Slim\DeferredCallable), Object(Slim\Http\Request), Object(Slim\Http\Response), Object(Closure))
#29 /var/www/public/api/v1/index.php(280): Slim\App->Slim\{closure}(Object(Slim\Http\Request), Object(Slim\Http\Response))
#30 [internal function]: Closure->{closure}(Object(Slim\Http\Request), Object(Slim\Http\Response), Object(Closure))
#31 /var/www/lib/slim/slim/Slim/DeferredCallable.php(57): call_user_func_array(Object(Closure), Array)
#32 [internal function]: Slim\DeferredCallable->__invoke(Object(Slim\Http\Request), Object(Slim\Http\Response), Object(Closure))
#33 /var/www/lib/slim/slim/Slim/MiddlewareAwareTrait.php(70): call_user_func(Object(Slim\DeferredCallable), Object(Slim\Http\Request), Object(Slim\Http\Response), Object(Closure))
#34 /var/www/public/api/v1/index.php(320): Slim\App->Slim\{closure}(Object(Slim\Http\Request), Object(Slim\Http\Response))
#35 [internal function]: Closure->{closure}(Object(Slim\Http\Request), Object(Slim\Http\Response), Object(Closure))
#36 /var/www/lib/slim/slim/Slim/DeferredCallable.php(57): call_user_func_array(Object(Closure), Array)
#37 [internal function]: Slim\DeferredCallable->__invoke(Object(Slim\Http\Request), Object(Slim\Http\Response), Object(Closure))
#38 /var/www/lib/slim/slim/Slim/MiddlewareAwareTrait.php(70): call_user_func(Object(Slim\DeferredCallable), Object(Slim\Http\Request), Object(Slim\Http\Response), Object(Closure))
#39 /var/www/public/api/v1/index.php(344): Slim\App->Slim\{closure}(Object(Slim\Http\Request), Object(Slim\Http\Response))
#40 [internal function]: Closure->{closure}(Object(Slim\Http\Request), Object(Slim\Http\Response), Object(Closure))
#41 /var/www/lib/slim/slim/Slim/DeferredCallable.php(57): call_user_func_array(Object(Closure), Array)
#42 [internal function]: Slim\DeferredCallable->__invoke(Object(Slim\Http\Request), Object(Slim\Http\Response), Object(Closure))
#43 /var/www/lib/slim/slim/Slim/MiddlewareAwareTrait.php(70): call_user_func(Object(Slim\DeferredCallable), Object(Slim\Http\Request), Object(Slim\Http\Response), Object(Closure))
#44 /var/www/lib/slim/slim/Slim/MiddlewareAwareTrait.php(117): Slim\App->Slim\{closure}(Object(Slim\Http\Request), Object(Slim\Http\Response))
#45 /var/www/lib/slim/slim/Slim/App.php(392): Slim\App->callMiddlewareStack(Object(Slim\Http\Request), Object(Slim\Http\Response))
#46 /var/www/lib/slim/slim/Slim/App.php(297): Slim\App->process(Object(Slim\Http\Request), Object(Slim\Http\Response))
#47 /var/www/public/api/v1/index.php(345): Slim\App->run()
#48 {main}
~~~

**Comment by hanjoosten** [comment]
However, After pressing again on the button to initialize the database, it succeeded. So there probably is some kind of timing issue (In an edge case?)

**Comment by Michiel-s** [comment]
I'll look into this again.

**Comment by Michiel-s** [comment]
Same as #88. Is fixed already in v1.14.0
